### PR TITLE
Feat : BaseResponse dto 추가

### PR DESCRIPTION
**Pull request by gombasan**
### 작업 사항
[feat(dto) : BaseResponse 클래스 추가](https://github.com/potenday-project/Clover/pull/36/commits/93b67eee2e3174999392b2cf8d66384b6dc552e2)
- api 가 반환 하는 타입을 하나의 dto 클래스로 관리하기 위하여 추가하였습니다.
- 사용자의 요청에 대한 코드와 메세지, 데이터를 묶어서 전달합니다.
- 기존 Response 보다 디테일 한 정보를 담기 위해 추가하였습니다.

[feat(dto) : ResponseType enum 클래스 추가](https://github.com/potenday-project/Clover/pull/36/commits/512e55c940e3c54913cbd56115aa11adadadf600)
- 기존 ErrorType 과 같은 목적으로 생성하였습니다.
- 각자 연관되어 있는 상수를 표현하기 위해 enum 클래스로 생성하였습니다.

[chore(Controller) : Controller 반환 타입 변경](https://github.com/potenday-project/Clover/pull/36/commits/f43181105375b50df68345f93f8126c79f6b6f07)
- 모든 Controller 메소드가 BaseResponse 타입을 반환하도록 변경하였습니다.
- 기존에는 단순한 데이터만 전달 하였기에, ErrorResponse 와 마찬가지로 디테일 한 정보를 같이 보내고자 했습니다.
- 컨트롤러에서 반환하는 데이터를 데이터부분, 메세지, 코드 구조로 반환하도록 변경했습니다.

[test(Controller) : ControllerTest 검증 부분 수정](https://github.com/potenday-project/Clover/pull/36/commits/fcc9c8242cd77b6f82c9e173666a6b89525a78b2)
- 기존 Controller 반환 타입이 변경되었고, 반환되는 데이터만 검증하므로,jsonPath 를 $.data 로 변경하였습니다.

### CODE REVIEW

**Comment by gombasan** [comment]
### 테스트 커버리지 리포트입니다
|File|Coverage [97.96%]|:green_apple:|
|:-|:-:|:-:|
|[HabitController.java](https://github.com/potenday-project/Clover/blob/512e55c940e3c54913cbd56115aa11adadadf600/src%2Fmain%2Fjava%2Fcom%2Fclover%2Fhabbittracker%2Fdomain%2Fhabit%2Fapi%2FHabitController.java)|100%|:green_apple:|
|[MemberController.java](https://github.com/potenday-project/Clover/blob/512e55c940e3c54913cbd56115aa11adadadf600/src%2Fmain%2Fjava%2Fcom%2Fclover%2Fhabbittracker%2Fdomain%2Fmember%2Fapi%2FMemberController.java)|100%|:green_apple:|
|[DiaryController.java](https://github.com/potenday-project/Clover/blob/512e55c940e3c54913cbd56115aa11adadadf600/src%2Fmain%2Fjava%2Fcom%2Fclover%2Fhabbittracker%2Fdomain%2Fdiary%2Fapi%2FDiaryController.java)|100%|:green_apple:|
|[ResponseType.java](https://github.com/potenday-project/Clover/blob/512e55c940e3c54913cbd56115aa11adadadf600/src%2Fmain%2Fjava%2Fcom%2Fclover%2Fhabbittracker%2Fglobal%2Fdto%2FResponseType.java)|100%|:green_apple:|
|[BaseResponse.java](https://github.com/potenday-project/Clover/blob/512e55c940e3c54913cbd56115aa11adadadf600/src%2Fmain%2Fjava%2Fcom%2Fclover%2Fhabbittracker%2Fglobal%2Fdto%2FBaseResponse.java)|88.89%|:green_apple:|

|Total Project Coverage|71.77%|:x:|
|:-|:-:|:-:|

File: src/main/java/com/clover/habbittracker/domain/diary/api/DiaryController.java [hunk 1]
```diff
@@ -29,25 +32,30 @@ public class DiaryController {
...
+		Long data = diaryService.register(memberId, request);
+		return new ResponseEntity<>(BaseResponse.of(data, DIARY_CREATE), HttpStatus.CREATED);
 	}
 
 	@GetMapping
-	ResponseEntity<List<DiaryResponse>> getMyDiaryList(Authentication authentication, @RequestParam(required = false) String date) {
-		Long memberId = (Long) authentication.getPrincipal();
-		return new ResponseEntity<>(diaryService.getMyList(memberId,date),HttpStatus.OK);
+	ResponseEntity<BaseResponse<List<DiaryResponse>>> getMyDiaryList(Authentication authentication,
+		@RequestParam(required = false) String date) {
```
Comment (JoosungKwon) [inline src/main/java/com/clover/habbittracker/domain/diary/api/DiaryController.java:43]: @AuthenticationPrincipal 도 고려해보세요~

**Review by JoosungKwon — APPROVED** [review]
@gombasan 고생하셨습니다~~ 잘하구 있어요 👍

**Comment by gombasan** [comment]
### 테스트 커버리지 리포트입니다
|File|Coverage [97.81%]|:green_apple:|
|:-|:-:|:-:|
|[HabitController.java](https://github.com/potenday-project/Clover/blob/90b1fa579e7dbba2f3a51767a871c40efdbfeef0/src%2Fmain%2Fjava%2Fcom%2Fclover%2Fhabbittracker%2Fdomain%2Fhabit%2Fapi%2FHabitController.java)|100%|:green_apple:|
|[MemberController.java](https://github.com/potenday-project/Clover/blob/90b1fa579e7dbba2f3a51767a871c40efdbfeef0/src%2Fmain%2Fjava%2Fcom%2Fclover%2Fhabbittracker%2Fdomain%2Fmember%2Fapi%2FMemberController.java)|100%|:green_apple:|
|[DiaryController.java](https://github.com/potenday-project/Clover/blob/90b1fa579e7dbba2f3a51767a871c40efdbfeef0/src%2Fmain%2Fjava%2Fcom%2Fclover%2Fhabbittracker%2Fdomain%2Fdiary%2Fapi%2FDiaryController.java)|100%|:green_apple:|
|[ResponseType.java](https://github.com/potenday-project/Clover/blob/90b1fa579e7dbba2f3a51767a871c40efdbfeef0/src%2Fmain%2Fjava%2Fcom%2Fclover%2Fhabbittracker%2Fglobal%2Fdto%2FResponseType.java)|100%|:green_apple:|
|[BaseResponse.java](https://github.com/potenday-project/Clover/blob/90b1fa579e7dbba2f3a51767a871c40efdbfeef0/src%2Fmain%2Fjava%2Fcom%2Fclover%2Fhabbittracker%2Fglobal%2Fdto%2FBaseResponse.java)|88.89%|:green_apple:|

|Total Project Coverage|71.53%|:x:|
|:-|:-:|:-:|

File: src/main/java/com/clover/habbittracker/domain/diary/api/DiaryController.java [hunk 2]
```diff
@@ -32,25 +32,23 @@ public class DiaryController {
 	private final DiaryService diaryService;
 
 	@PostMapping
-	ResponseEntity<BaseResponse<Long>> createDiary(Authentication authentication, @RequestBody DiaryRequest request) {
-		Long memberId = (Long)authentication.getPrincipal();
-		Long data = diaryService.register(memberId, request);
-		return new ResponseEntity<>(BaseResponse.of(data, DIARY_CREATE), HttpStatus.CREATED);
+	ResponseEntity<BaseResponse<Long>> createDiary(@AuthenticationPrincipal Long memberId, @RequestBody DiaryRequest request) {
+		Long registerId = diaryService.register(memberId, request);
+		return new ResponseEntity<>(BaseResponse.of(registerId, DIARY_CREATE), HttpStatus.CREATED);
```
Comment (JoosungKwon) [inline src/main/java/com/clover/habbittracker/domain/diary/api/DiaryController.java:37]:
``` suggestion
                Long registerId = diaryService.register(memberId, request);
                BaseResponse response = BaseResponse.of(registerId, DIARY_CREATE);
		return ResponseEntity.ok().body(response);
```
이렇게 할수 도 있다는걸 제안드려봅니다~~

**Comment by gombasan** [comment]
### 테스트 커버리지 리포트입니다
|File|Coverage [97.86%]|:green_apple:|
|:-|:-:|:-:|
|[HabitController.java](https://github.com/potenday-project/Clover/blob/a403e55a9c03c930dcac3153cce6c4649267de3f/src%2Fmain%2Fjava%2Fcom%2Fclover%2Fhabbittracker%2Fdomain%2Fhabit%2Fapi%2FHabitController.java)|100%|:green_apple:|
|[MemberController.java](https://github.com/potenday-project/Clover/blob/a403e55a9c03c930dcac3153cce6c4649267de3f/src%2Fmain%2Fjava%2Fcom%2Fclover%2Fhabbittracker%2Fdomain%2Fmember%2Fapi%2FMemberController.java)|100%|:green_apple:|
|[DiaryController.java](https://github.com/potenday-project/Clover/blob/a403e55a9c03c930dcac3153cce6c4649267de3f/src%2Fmain%2Fjava%2Fcom%2Fclover%2Fhabbittracker%2Fdomain%2Fdiary%2Fapi%2FDiaryController.java)|100%|:green_apple:|
|[ResponseType.java](https://github.com/potenday-project/Clover/blob/a403e55a9c03c930dcac3153cce6c4649267de3f/src%2Fmain%2Fjava%2Fcom%2Fclover%2Fhabbittracker%2Fglobal%2Fdto%2FResponseType.java)|100%|:green_apple:|
|[BaseResponse.java](https://github.com/potenday-project/Clover/blob/a403e55a9c03c930dcac3153cce6c4649267de3f/src%2Fmain%2Fjava%2Fcom%2Fclover%2Fhabbittracker%2Fglobal%2Fdto%2FBaseResponse.java)|88.89%|:green_apple:|

|Total Project Coverage|71.41%|:x:|
|:-|:-:|:-:|

**Pull request merged**